tests/serac_thermal_solver.cpp

```diff
@@ -91,6 +92,153 @@ TEST(thermal_solver, static_solve)
   MPI_Barrier(MPI_COMM_WORLD);
 }
 
+TEST(thermal_solver, static_solve_multiple_bcs)
+{
+  MPI_Barrier(MPI_COMM_WORLD);
+
+  // mesh
+  std::string base_mesh_file = std::string(SERAC_SRC_DIR) + "/data/star_with_2_bdr_attributes.mesh";
+  const char* mesh_file      = base_mesh_file.c_str();
+
+  // Open the mesh
+  std::fstream imesh(mesh_file);
+  auto         mesh = std::make_unique<mfem::Mesh>(imesh, 1, 1, true);
+  imesh.close();
+
+  // Refine in serial
+  mesh->UniformRefinement();
+
+  // Initialize the parallel mesh and delete the serial mesh
+  auto pmesh = std::make_shared<mfem::ParMesh>(MPI_COMM_WORLD, *mesh);
+
+  // Refine the parallel mesh
+  pmesh->UniformRefinement();
+
+  // Initialize the second order thermal solver on the parallel mesh
+  ThermalSolver therm_solver(2, pmesh);
+
+  // Set the time integration method
+  therm_solver.SetTimestepper(TimestepMethod::QuasiStatic);
+
+  // Initialize the temperature boundary condition
+  auto u_0 = std::make_shared<mfem::FunctionCoefficient>(BoundaryTemperature);
+  auto u_1 = std::make_shared<mfem::ConstantCoefficient>(0.0);
+
+  std::vector<int> marked_1(pmesh->bdr_attributes.Max(), 0);
+  std::vector<int> marked_2(pmesh->bdr_attributes.Max(), 0);
+  marked_1[0] = 1;
```

Comment:
This now sets the temperature using a coefficient on two disjoint parts of the boundary. I've actually been considering having this data structure as a `std::set` of attributes instead of a `std::vector` of indicators and translate it to an `mfem::Array` in the solver routines.